browser/ui/sidebar/sidebar_utils.cc

```diff
@@ -50,20 +52,58 @@ bool IsActiveTabAlreadyAddedToSidebar(Browser* browser) {
 
 }  // namespace
 
+bool HiddenDefaultSidebarItemsContains(SidebarService* service,
+                                       const GURL& url) {
+  const auto not_added_default_items = service->GetHiddenDefaultSidebarItems();
+  if (not_added_default_items.empty())
+    return false;
+  const GURL converted_url = ConvertURLToBuiltInItemURL(url);
+  for (const auto& item : not_added_default_items) {
+    if (item.url == converted_url)
+      return true;
+  }
+  return false;
+}
+
 bool CanUseSidebar(Browser* browser) {
   DCHECK(browser);
   return browser->is_type_normal();
 }
 
+// If url is relavant with bulitin items, use builtin item's url.
+// Ex, we don't need to add bookmarks manager as a sidebar shortcut
+// if sidebar panel already has bookmarks item.
+GURL ConvertURLToBuiltInItemURL(const GURL& url) {
+  if (url == GURL(chrome::kChromeUIBookmarksURL))
+    return GURL(kSidebarBookmarksURL);
+
+  if (url.host() == "talk.brave.com")
```

Comment:
nit: maybe this can be a constant (in the same file as `kBraveTalkURL`). Like `kBraveTalkHost`

Reply:
Done!